npc/custom/warper.txt

```diff
@@ -149,42 +155,45 @@ T14: Go("gonryun",160,120);
 T15: Go("hugel",96,145);
 T16: Go("izlude",128,(checkre(3)?146:114));
 T17: Go("jawaii",251,132);
-T18: Go("lighthalzen",158,92);
-T19: Go("louyang",217,100);
-T20: Go("xmas",147,134);
-T21: Restrict("RE");
-	 Go("malangdo",140,114);
+T18: Restrict("RE");
+	 Go("lasagna",193,182);
+T19: Go("lighthalzen",158,92);
+T20: Go("louyang",217,100);
+T21: Go("xmas",147,134);
 T22: Restrict("RE");
+	 Go("malangdo",140,114);
+T23: Restrict("RE");
 	 Go("malaya",231,200);
-T23: Go("manuk",282,138);
-T24: Go("mid_camp",210,288);
-T25: Restrict("RE");
+T24: Go("manuk",282,138);
+T25: Go("mid_camp",210,288);
+T26: Restrict("RE");
 	 Go("mora",55,146);
-T26: Go("morocc",156,93);
-T27: Go("moscovia",223,184);
-T28: Go("nameless_n",256,215);
-T29: Go("niflheim",202,174);
-T30: Go("payon",179,100);
-T31: Go("rachel",130,110);
-T32: Go("splendide",201,147);
-T33: Go("thor_camp",246,68);
-T34: Go("umbala",97,153);
-T35: Go("veins",216,123);
-T36: Go("yuno",157,51);
+T27: Go("morocc",156,93);
+T28: Go("moscovia",223,184);
+T29: Go("nameless_n",256,215);
+T30: Go("niflheim",202,174);
+T31: Go("payon",179,100);
+T32: Go("rachel",130,110);
+T33: Go("splendide",201,147);
+T34: Go("thor_camp",246,68);
+T35: Go("umbala",97,153);
+T36: Go("veins",216,123);
+T37: Go("yuno",157,51);
 
 // --------------------------------------------------
 	Fields:
 // --------------------------------------------------
+@f = true;
```

Comment:
I know it's the easiest solution. But most of the time you should not use char-variables in an NPC if you only need the variable in the same scope. Another solution to that would be for example another parameter for `Pick()` or another function like `PickDungeon()` (Which just calls Pick() with the dungeon parameter).

Anyway since the whole NPC is built on similar behavior (`setarray @c`, `@warp_block`), it should be fine. I just wanted to mention it for your future projects.